Nightly K8s Matrix CI 2024-04-06: Failed

action url: https://github.com/spidernet-io/spiderpool/actions/runs/8583491067

```
nodes:
  - role: control-plane
  - role: worker
KIND_OPTION="" ; \
       		[ -n "v1.26.2" ] && KIND_OPTION=" --image docker.io/kindest/node:v1.26.2 " && echo "setup kind with docker.io/kindest/node:v1.26.2"; \
            kind create cluster --config /home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool0406201327/kind-config.yaml \
		--name spiderpool0406201327 --kubeconfig /home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool0406201327/.kube/config ${KIND_OPTION}
/etc/bash.bashrc: line 7: PS1: unbound variable
setup kind with docker.io/kindest/node:v1.26.2
Creating cluster "spiderpool0406201327" ...
 • Ensuring node image (docker.io/kindest/node:v1.26.2) 🖼  ...
 ✓ Ensuring node image (docker.io/kindest/node:v1.26.2) 🖼
 • Preparing nodes 📦 📦   ...
 ✗ Preparing nodes 📦 📦 
Deleted nodes: ["spiderpool0406201327-control-plane" "spiderpool0406201327-worker"]
ERROR: failed to create cluster: command "docker run --name spiderpool0406201327-control-plane --hostname spiderpool0406201327-control-plane --label io.x-k8s.kind.role=control-plane --privileged --security-opt seccomp=unconfined --security-opt apparmor=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro -e KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER --detach --tty --label io.x-k8s.kind.cluster=spiderpool0406201327 --net kind --restart=on-failure:1 --init=false --cgroupns=private --sysctl=net.ipv6.conf.all.disable_ipv6=0 --sysctl=net.ipv6.conf.all.forwarding=1 --publish=127.0.0.1:46635:6443/TCP -e KUBECONFIG=/etc/kubernetes/admin.conf docker.io/kindest/node:v1.26.2" failed with error: exit status 125

Command Output: 322e9b477cff56b4c04fbb0f260225a4d4fa6a964e3f713984ebe3722ff215ad
docker: Error response from daemon: driver failed programming external connectivity on endpoint spiderpool0406201327-control-plane (3c0e458ccc57583f619c9a0a668b910865607d0274060dfdd1c3a029a1f0c598): fork/exec /usr/bin/docker-proxy: exec format error.
make[3]: *** [Makefile:142: setup_kind] Error 1
make[3]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make[2]: *** [Makefile:31: kind-init] Error 2
make[2]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make[1]: *** [Makefile:297: e2e_init] Error 2
make[1]: Leaving directory '/home/runner/work/spiderpool/spiderpool'
make: *** [Makefile:321: e2e_init_underlay] Error 2
```

--- a/.github/workflows/e2e-init.yaml
+++ b/.github/workflows/e2e-init.yaml
@@ -134,10 +134,16 @@ jobs:
       # test against commit version
       # https://github.com/kubernetes-sigs/kind/issues/2863
       - name: Setup Kind Cluster
-        uses: nick-invision/retry@v2
+        uses: nick-invision/retry@v3
         with:
           timeout_minutes: 20
           max_attempts: 3
+          on_retry_command: |
+            echo "debug"
+            systemctl status docker
+            journalctl -xeu docker
+            echo "restart docker before trying again"
+            systemctl restart docker
           command: |
             INSTALL_OVS_VALUE=true
             MINIMAL_VERSION=1.24.3

